bootstrap

```diff
@@ -29,6 +29,7 @@ virtualenv -p python3 --system-site-packages --distribute virtualenv
 
 # avoid pip bugs
 ./virtualenv/bin/pip install --upgrade pip
+pip3 install --upgrade setuptools cffi  # address pip issue: https://github.com/pypa/pip/issues/6264
```

Comment:
this should probably use the same `./virtualenv/bin/pip` instead of `pip3` right?

Reply:
the problem is related to pip & venv build isolation, an update to the external python setuptools module solves the error.